uberduck_ml_dev/models/tacotron2.py

```diff
@@ -424,6 +424,79 @@ def inference_noattention(self, memory, attention_map):
 
         return mel_outputs, gate_outputs, alignments
 
+    def inference_partial_tf(self, memory, decoder_inputs, tf_until_idx):
+        """Decoder inference with teacher-forcing up until tf_until_idx
+        PARAMS
+        ------
+        memory: Encoder outputs
+        decoder_inputs: Decoder inputs for teacher forcing. i.e. mel-specs
+        memory_lengths: Encoder output lengths for attention masking.
+
+        RETURNS
+        -------
+        mel_outputs: mel outputs from the decoder
+        gate_outputs: gate outputs from the decoder
+        alignments: sequence of attention weights from the decoder
+        """
+        if torch.cuda.is_available():
+            decoder_inputs = decoder_inputs.cuda()
+
+        B = memory.size(0)
+        decoder_inputs = self.parse_decoder_inputs(decoder_inputs)
+        decoder_inputs = decoder_inputs.reshape(
+            -1, decoder_inputs.size(1), self.n_mel_channels
+        )
+        decoder_input = self.get_go_frame(memory).unsqueeze(0)
+        decoder_inputs = torch.cat((decoder_input, decoder_inputs), dim=0)
+        decoder_inputs = self.prenet(decoder_inputs)
+
+        self.initialize_decoder_states(memory, mask=None)
+
+        mel_outputs = torch.empty(
+            B, 0, self.n_frames_per_step_current * self.n_mel_channels
+        )
+        if torch.cuda.is_available():
```

Comment:
Can we add an additional check to make sure we want to use the gpu? This will cause tests to fail locally if run on a machine with a GPU